cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/transform/ScriptTransform.java

```diff
@@ -167,31 +184,46 @@ private StructuredRecord decodeRecord(Map nativeObject, Schema schema) {
     return builder.build();
   }
 
+  private List jsObject2List(Object object) {
+    if (object.getClass().getName().equals("jdk.nashorn.api.scripting.ScriptObjectMirror")) {
```

Comment:
If we assume we only support Java 7 and 8, then a non-null `somValuesMethod` would mean the object will be a ScriptObjectMirror.